src/core/manager-serialize.c

```diff
@@ -529,6 +532,14 @@ int manager_deserialize(Manager *m, FILE *f, FDSet *fds) {
                                 log_notice("Failed to parse soft reboots counter '%s', ignoring.", val);
                         else
                                 m->soft_reboots_count = n;
+                } else if ((val = startswith(l, "objective="))) {
+                        unsigned n;
+
+                        if (safe_atou(val, &n) < 0 || n > _MANAGER_OBJECTIVE_MAX)
+                                log_notice("Failed to parse objective '%s', ignoring.", val);
+                        else
+                                m->objective = n;
```

Comment:
hmm, this doesn't sound right. the objective tells us what we should do next. After coming back from a soft reboot the objective is no longer is to soft reboot however. 

Hence, this should not be deserialized into the `objective` field again, but there should be a new field `prior_object` or so which stores this info.

i think the log output at boot should output the previous objective btw somewhere in its early output. i.e. some log message that says "Initializing after previous service manager objective '%s'." or so.

Reply:
Given this is really only taken across on softreboot the log is not that useful right now so I'll leave that out, can be added later if it turns out to be useful